### DISCUSSION TITLE
`make` colour scheme not well readable on light background. Wish: Options to specify colour scheme and/ or autodetect.

### DISCUSSION THREAD

**Discussion by dreirund** [Freetz-NG/freetz-ng#1039] [Unerfüllte Wünsche]
Ahoj,

Freetz-NG's build system has coloured output with some colours that are barely readable on black-on-white terminals:

---

![Freetz-NFG_make-screenshot](https://github.com/user-attachments/assets/af60214e-725f-42ad-b198-76c4efefad22)

---

I have the feature-wish to either autodetect colour scheme (which I think is not easily possible), or provide options

* to specify a colour scheme suitable for
  - light background,
  - dark background,
* not doing any colouring at all.

Regards!

**Comment by fda77** [comment]
The easiest solution is to set the background color during make to eye friendly black ;)

**Reply by dreirund** [reply]
> *eye friendly black*

"Eye friendly" is dependent on the circumstances; and I doubt that there are any health issues associated with it.

When the surroundings are bright as well, then bright background is more comfortable in my experience (and using bright surroundings during day is better for the ciradian rythm anyway).

So, I doubt the general judgement "black is more eye friendly".

> *The easiest solution is to set the background color during make to [...] black*

A solution which requires from the user to change the own workplace is not a solution but a workaround. (Fair enough, sometimes workarounds are the only feasible way to go, but "solution" is the wron term here in my opinion.)

Regards!

**Reply by fda77** [reply]
No, as i wrote 
> set the background color during make

and not >before< make

eg   `printf %b '\e[41m' '\e[8]' '\e[H\e[J'`

**Comment by dreirund** [comment]
On Tue, 10 Sep 2024 13:54:12 -0700, fda77 ***@***.***>
wrote:

> No, as i wrote 
> > set the background color during make  
> 
> and not >before< make
> 
> eg   `printf %b '\e[41m' '\e[8]' '\e[H\e[J'`

Ah OK, so writing a wrapper that sends ANSI terminal colour escape
codes.

Yes, this could be another workaround, thanks!

**Reply by fda77** [reply]
its the same way the foreground color are colored.
